tracking-templates/tracking.js

```diff
@@ -68,6 +73,43 @@
       if (g._log) g._log(LOG_EVENT_TYPE.HB_REQ);
     } catch(e) {}
   };
+  var lsAvailable = !!window.localStorage && !!window.localStorage.setItem && !!window.localStorage.getItem;
+  function serializeSessionEnds(sessionEnds, maxLength = 100) {
+    var sids = Object.keys(sessionEnds);
+    var start_idx = sids.length > maxLength ? maxLength - sids.length : 0;
+    var serialized = '';
+    for (var i = start_idx; i < sids.length; i++) {
+      serialized = serialized + sids[i] + '=' + sessionEnds[sids[i]] + ',';
+    }
+    return serialized.slice(0, -1);
+  }
+  function deserializeSessionEnds(sessionEndsString) {
+    if (!sessionEndsString) {
+      return {}
+    }
+    var sessionEndEntries = sessionEndsString.split(',');
+    var deserialized = {};
+    for (var i = 0; i < sessionEndEntries.length; i++) {
+      var split = sessionEndEntries[i].split('=');
+      deserialized[split[0]] = split[1]
+    }
+    return deserialized;
+  }
+  g._asUpdate = function (s) {
+    if (!lsAvailable) return;
+    var sid = typeof s !== 'undefined' ? s : '{{SESSION_ID}}';
```

Comment:
because my understandin is that, when the `/new.js` script on `switchChannel` or manual init is called, this script gets a new SESSION_ID set server-side (included in the `HEARTBEAT_URL` variable), and `g._hb` is updated so send heartbeats with the right session-id. To also save the timestamps for this new session correctly, I have to call the `g._asUpdate` method with the new session-id as param, as `g._asUpdate` is defined in the `tracking.js` template (used by `ra.js` and `ra_if.js`), which is not reloaded, and therefore the previous/first session-id would always be used.